test/meetup_cache_worker_test.exs

```diff
@@ -15,127 +14,24 @@ defmodule MeetupBot.MeetupCacheWorkerTest do
-                  }
-                }
-              ]
-            }
-          }
-        }
+  test "perform/1 fetches and stores new meetup events" do
+    stub_meetup_response([
+      %{
+        id: "123",
+        title: "Testing with ExUnit",
+        eventUrl: "http://example.com",
+        dateTime: tomorrow(),
+        endTime: day_after_tomorrow()
       }
-      """)
-    end)
-
-    Bypass.expect(bypass_gdg(), "GET", "/api/event", fn conn ->
-      conn
-      |> Plug.Conn.put_resp_content_type("application/json")
-      |> Plug.Conn.resp(200, """
-      {
-        "results": [
-          {
-            "id": 456,
-            "title": "Google Cloud Platform",
-            "url": "http://gdg.example.com",
-            "start_date": "2024-03-29T19:00:00",
-            "end_date": "2024-03-29T21:00:00"
-          }
-        ]
-      }
-      """)
-    end)
+    ])
 
     assert :ok = MeetupCacheWorker.perform(%Oban.Job{})
 
-    assert [meetup, gdg] = MeetupCache.all()
-
-    assert meetup.source_id == "123"
-    assert meetup.name == "Elixir Meetup"
-
-    assert gdg.source_id == "456"
-    assert gdg.name == "GDG"
-  end
-
-  test "perform/1 update existings meetups", %{} do
-    %Event{
-      source: "meetup",
-      source_id: "123",
-      datetime: ~N[2024-03-28 19:00:00]
-    }
-    |> Repo.insert!()
-
-    %Event{
-      source: "GDG",
-      source_id: "456",
-      datetime: ~N[2024-03-30 19:00:00]
-    }
-    |> Repo.insert!()
-
-    Bypass.expect(bypass_meetup(), "POST", "/gql", fn conn ->
-      conn
-      |> Plug.Conn.put_resp_content_type("application/json")
-      |> Plug.Conn.resp(200, """
-      {
-        "data": {
-          "g0": {
-            "name": "Elixir Meetup",
-            "upcomingEvents": {
-              "edges": [
-                {
-                  "node": {
-                    "id": "123",
-                    "title": "Testing with ExUnit",
-                    "eventUrl": "http://example.com",
-                    "dateTime": "2024-03-29T19:00-03:00",
-                    "endTime": "2024-03-29T23:00-03:00"
-                  }
-                }
-              ]
-            }
-          }
-        }
-      }
-      """)
-    end)
-
-    Bypass.expect(bypass_gdg(), "GET", "/api/event", fn conn ->
-      conn
-      |> Plug.Conn.put_resp_content_type("application/json")
-      |> Plug.Conn.resp(200, """
-      {
-        "results": [
-          {
-            "id": 456,
-            "title": "Google Cloud Platform",
-            "url": "http://gdg.example.com",
-            "start_date": "2024-03-31T19:00:00",
-            "end_date": "2024-03-31T23:00:00"
-          }
-        ]
-      }
-      """)
-    end)
-
-    assert :ok = MeetupCacheWorker.perform(%Oban.Job{})
+    events = MeetupCache.all()
+    assert length(events) == 1
 
-    [meetup, gdg] = MeetupCache.all()
-    assert meetup.datetime == ~N[2024-03-29 19:00:00]
-    assert gdg.datetime == ~N[2024-03-31 19:00:00]
```

Comment:
I would only add one more test to verify that the upcoming events not present in source are deleted.